### PR TITLE
fix(webpack-loader): support json catalogs. Improve coverage

### PR DESCRIPTION
**Pull request by thekip**
# Description

Using json catalogs with webpack loader was broken. Fixed that and added that case to a tests

Details overview: 

- When webpack encounter a json file it automatically set a `json` content type for that module. Despite the fact the loader may return a different type of file. There is no way (except of dirty hacks involving executing private internals of webpack) to tell to webpack that this loader will perform a transformation from `json` to `js`.  So intstead i just compile catalog in json format and feed it to webpack. 

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

Fixes # (issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thekip** [comment]
Blocked by build pipeline. Release should be build before running this test. I will look for solutions

**Comment by thekip** [comment]
Ready. Windows WF failed yesterday because of no binary for the latest node was precompiled. It seems today they fixed it.

File: website/docs/ref/loader.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # Webpack Loader
 
-It's a good practice to use compiled message catalogs during development. However, running [`compile`](/docs/ref/cli.md#compile) everytime messages are changed soon becomes tedious.
-
-`@lingui/loader` is a webpack loader, which compiles messages on the fly:
+Webpack loader which compiles catalogs on the fly. In summary, `lingui compile` command isn't required when using this loader
```
Comment (andrii-bodnar) [inline website/docs/ref/loader.md:3]:
```suggestion
The Webpack loader compiles catalogs on the fly. In summary, the `lingui compile` command isn't needed when using this loader.
```

File: website/docs/ref/loader.md [hunk 2]
```diff
@@ -16,14 +14,21 @@ npm install --save-dev @lingui/loader
 
 Simply prepend `@lingui/loader!` in front of path to message catalog you want to import. Here's an example of dynamic import:
 
-Extension is mandatory. If you use minimal or lingui file format, use `.json`. In case of using po format, use `.po`.
-
-``` jsx
+Extension is mandatory.
```
Comment (andrii-bodnar) [inline website/docs/ref/loader.md:17]:
```suggestion
The extension is mandatory.

```

File: website/docs/ref/loader.md [hunk 3]
```diff
@@ -16,14 +14,21 @@ npm install --save-dev @lingui/loader
...
 export async function dynamicActivate(locale: string) {
-   const { messages } = await import(`@lingui/loader!./locales/${locale}/messages.json`)
+   const { messages } = await import(`@lingui/loader!./locales/${locale}/messages.po`)
    i18n.load(locale, messages)
    i18n.activate(locale)
 }
 ```
 
+:::note
+Catalogs with `.json` extension treated differently by Webpack. They loaded as ES module with default export, so your import should look like that:
```
Comment (andrii-bodnar) [inline website/docs/ref/loader.md:27]:
```suggestion
Catalogs with the `.json` extension are treated differently by Webpack. They load as ES module with default export, so your import should look like this:
```

File: website/docs/ref/loader.md [hunk 4]
```diff
@@ -16,14 +14,21 @@ npm install --save-dev @lingui/loader
...
+
+```ts
+const { messages } = (await import(`@lingui/loader!./locales/${locale}/messages.po`)).default
```
Comment (andrii-bodnar) [inline website/docs/ref/loader.md:30]: is the extension correct?

Comment (thekip) [reply website/docs/ref/loader.md]: thanks, fixed

**Pull request merged**